app/waf/src/services/asg.service.ts

```diff
@@ -312,26 +336,58 @@ export class ASGManager {
     );
   }
 
-  async icontrolGet(trustDeviceId: string, path: string): Promise<object> {
+  async installCert(trustDeviceId: string, body: object): Promise<object> {
+    return await this.icontrolPost(
+      trustDeviceId,
+      '/mgmt/tm/sys/crypto/cert',
+      body,
+    );
+  }
+  async installKey(trustDeviceId: string, body: object): Promise<object> {
+    return await this.icontrolPost(
+      trustDeviceId,
+      '/mgmt/tm/sys/crypto/key',
+      body,
+    );
+  }
+
+  async uploadFile(
+    trustDeviceId: string,
+    content: string,
+    length: number,
+    name: string,
+  ): Promise<object> {
+    return await this.service.uploadFile(
+      ASG_HOST,
+      ASG_PORT,
+      trustDeviceId,
+      `/mgmt/shared/file-transfer/uploads/${name}`,
+      length - 1,
+      length,
+      content,
+    );
+  }
+
+  async icontrolGet(trustDeviceId: string, url: string): Promise<object> {
     return await this.service.icontrolGet(
       ASG_HOST,
       ASG_PORT,
       trustDeviceId,
-      path,
+      url,
     );
   }
 
   async icontrolPost(
     trustDeviceId: string,
-    path: string,
+    url: string,
```

Comment:
Do you need to rename this parameter?

Reply:
this is due to a conflict during the merge.  I will double check that.